app/controllers/api/vcs.php

```diff
@@ -593,32 +611,103 @@
-            ->addDetector(new CPP())
-            ->addDetector(new Deno())
-            ->addDetector(new Dotnet());
+            $frameworks = Config::getParam('frameworks');
+            $frameworkDetail = \array_reverse(\array_filter(\array_keys($frameworks), function ($key) use ($framework, $frameworks) {
+                return $frameworks[$key]['key'] === $framework;
+            }))[0] ?? '';
+            $detection->setAttribute('framework', $frameworkDetail);
 
-        $runtime = $detectorFactory->detect();
+            $response->dynamic($detection, Response::MODEL_FRAMEWORK_DETECTION);
+        } else {
+            $detection = new Document([
+                'runtime' => '',
+                'commands' => '',
+                'entrypoint' => '',
+            ]);
 
-        $runtimes = Config::getParam('runtimes');
-        $runtimeDetail = \array_reverse(\array_filter(\array_keys($runtimes), function ($key) use ($runtime, $runtimes) {
-            return $runtimes[$key]['key'] === $runtime;
-        }))[0] ?? '';
+            $strategies = [
+                new Strategy(Strategy::FILEMATCH),
+                new Strategy(Strategy::LANGUAGES),
+                new Strategy(Strategy::EXTENSION),
+            ];
+
+            foreach ($strategies as $strategy) {
+                $runtimeDetector = new Runtime($strategy === Strategy::LANGUAGES ? $languages : $files, $strategy, $packagerName);
+                $runtimeDetector
+                    ->addOption(new Node())
+                    ->addOption(new Bun())
+                    ->addOption(new Deno())
+                    ->addOption(new PHP())
+                    ->addOption(new Python())
+                    ->addOption(new Dart())
+                    ->addOption(new Swift())
+                    ->addOption(new Ruby())
+                    ->addOption(new Java())
+                    ->addOption(new CPP())
+                    ->addOption(new Dotnet());
+
+                $detectedRuntime = $runtimeDetector->detect();
+
+                if ($detectedRuntime) {
+                    $detection->setAttribute('commands', $detectedRuntime->getCommands());
+                    $detection->setAttribute('entrypoint', $detectedRuntime->getEntrypoint());
+                    $runtime = $detectedRuntime->getName();
+                    break;
+                }
+            }
 
-        $detection = [];
-        $detection['runtime'] = $runtimeDetail;
+            if (!empty($runtime)) {
+                $runtimes = Config::getParam('runtimes');
+                $runtimeDetail = \array_reverse(\array_filter(\array_keys($runtimes), function ($key) use ($runtime, $runtimes) {
+                    return $runtimes[$key]['key'] === $runtime;
+                }))[0] ?? '';
+                $detection->setAttribute('runtime', $runtimeDetail);
+            } else {
+                throw new Exception(Exception::GENERAL_BAD_REQUEST, 'Runtime not detected.');
```

Comment:
Let's have proper error type for this, avoid using gneric GENERAL_BAD_REQUEST as rule of thumb.